vignettes/Introduction.Rmd

```diff
@@ -594,35 +594,29 @@ pq <- PrettyCountingQueue$new(1, 2, "foobar")
 pq
 ```
 
-
 ### Finalizers
 
-Sometimes it's useful to run a function when the object is garbage collected. For example, you may want to make sure a file or database connection gets closed. To do this, you can call the `reg.finalizer()` function in your `initialize` method, and pass it `self` as the object that will trigger the function when it is garbage collected.
+Sometimes it's useful to run a function when the object is garbage collected. For example, you may want to make sure a file or database connection gets closed. To do this, you can define a `finalize()` method, which will be called with no arguments when the object is garbage collected.
 
 
 ```{r}
 A <- R6Class("A", public = list(
-  initialize = function() {
-    reg.finalizer(self,
-                  function(e) print("Finalizer has been called!"),
-                  onexit = TRUE)
+  finalize = function() {
+    print("Finalizer has been called!")
   }
 ))
 
 
-# Instantiate an object, but don't save a reference to it
-A$new()
-
-# Run something else to get rid of any references to the object, because the
-# last returned value is saved in .Last.value
-1+1
+# Instantiate an object:
+O <- A$new()
```

Comment:
Can you rename `O` to `a`, or `obj`? The name `O` is surprising and sort of distracts from the important parts.